services/gateway/gateway-service.mjs

```diff
@@ -339,11 +373,54 @@ class GatewayService {
       console.log('player has no last time record')
       return false;
     }
-    if( Date.now() - playerData.lastMovingTime < moveingRequestThreshold){
+    if( Date.now() - playerData.lastMovingTime < movingRequestThreshold){
       return false;
     }
     return true;
   }
+
+  /**
+   * Mark the cell specified by mapCoord as occupied by playerID.
+   * Note: This is atomic.
+   * @return {Boolean} success - Return true if the cell is marked
+   * successfully, false if it's occupied.
+   */
+  async _occupyCoord(mapCoord, playerID) {
+    let ret = await this.dir.getRedis().setAsync(
+      [mapCoord.toRedisKey(), playerID, 'NX']);
+    if (ret === null) {
+      // Failed
+      return false;
+    }
+    if (ret === 'OK') {
+      return true;
+    }
+    console.assert(`Invalid reply from redis for _occupyCoord: ${ret}`);
+    return false;
+  }
```

Comment:
Is `getRedis().setAsync()` going to return any value other than `OK` and `null`? I think you should find the documentation to ensure what value it is going to return.
```
const ret = await ...();
return (ret === 'OK');
```

Reply:
Generally it doesn't. The assertion is there to make sure it doesn't, and if it does due to any reasons, we'll know.

Reply:
Isn't the documentation of Redis lists all possible return values?